components/tidb_query_expr/src/impl_string.rs

```diff
@@ -499,6 +499,22 @@ pub fn upper(arg: BytesRef, writer: BytesWriter) -> Result<BytesGuard> {
     Ok(writer.write_ref(Some(arg)))
 }
 
+#[rpn_fn(writer)]
+#[inline]
+pub fn lower(arg: BytesRef, writer: BytesWriter) -> Result<BytesGuard> {
+    match str::from_utf8(arg) {
+        Ok(s) => Ok(writer.write_ref(Some(s.to_lowercase().as_bytes()))),
+        Err(err) => Err(box_err!("invalid input value: {:?}", err)),
+    }
+}
+
+#[rpn_fn(writer)]
+#[inline]
+pub fn lower_binary(arg: BytesRef, writer: BytesWriter) -> Result<BytesGuard> {
```

Comment:
`lower` and `lower_utf8` are different funtions, take `upper` and `upper_utf-8` a look maybe?

Reply:
Ah sure yes! I wasn't aware of that because in the old coprocessor and in TiDB there was only `lower`, but not `lower_utf8`. Is `lower()` also just a noop?

Reply:
> Is lower() also just a noop?

Yes!